lib/class-wp-block.php

```diff
@@ -150,13 +138,41 @@ function( $inner_block ) use ( $child_context, $registry ) {
 		}
 	}
 
+	/**
+	 * Returns a value from an inaccessible property.
+	 *
+	 * This is used to lazily initialize the `attributes` property of a block,
+	 * such that it is only prepared with default attributes at the time that
+	 * the property is accessed. For all other inaccessible properties, a `null`
+	 * value is returned.
+	 *
+	 * @param string $name Property name.
+	 *
+	 * @return array|null Prepared attributes, or null.
+	 */
+	public function __get( $name ) {
+		if ( 'attributes' === $name && ! isset( $this->attributes ) ) {
```

Comment:
Why do we need to check if `isset`?

Reply:
> Why do we need to check if `isset`?

It might not be necessary. In early iterations, I assumed `__get` was always called for all property access, but it turns out that it's only called for inaccessible properties. I assume it should be fine to remove the `isset` check if we're assigning a value to `$this->attributes` as part of the condition body.

Reply:
> Why do we need to check if `isset`?

Removed in c3e0b9d9a.